src/shared/components/palette/base-field/field-read.component.ts

```diff
@@ -39,7 +33,7 @@ export class FieldReadComponent extends AbstractFieldReadComponent implements On
     let component = this.resolver.resolveComponentFactory(componentClass).create(injector);
 
     // Provide component @Inputs
-    component.instance['caseField'] = this.caseField;
+    component.instance['caseField'] = plainToClassFromExist(new CaseField(), this.caseField);
```

Comment:
where is this.caseField coming from? wouldn't it be better to apply plainToClassFromExist at the origin rather than here?

Reply:
as you noticed, the field is coming from:
```<table *ngIf="caseField.value && caseField.value.length" class="collection-field-table">
  <tbody>
    <tr *ngFor="let item of caseField.value; let i = index">
      <td>
        <ccd-field-read [caseField]="{
          id: i,
          label: caseField.label + ' ' + (i + 1),
          field_type: caseField.field_type.collection_field_type,
          value: item.value,
          hidden: caseField.hidden
        }" [context]="context"></ccd-field-read>
      </td>
    </tr>
  </tbody>
</table>
```

We should refactor there. Various solutions we can pick the easier. You could create a method in the component that creates the field and call the method in the html.
Or see if it's possible to use plainToClassFromExist or a variant in the html. 
Or use new Field in the html 
etc...

Reply:
It looks like Mohammed did try the above suggestions of mine, but it's not that easy to make it work. So considering we need to release this because of time constraints, we can accept the proposed workaround